### PR TITLE
generate default implementation for an enum from an enum variant #6860

### LINKED ISSUE #6860: Feature Request: Derive Default from Enum Variant
A relatively common situation for me is an enum that has one variant that should become the default. It would be nice if one could select the variant that should become the default and be able to automatically derive a `Default` for it:

```rust
enum Variant {
    Undefined, // hover and "derive Default for this variant"
    Minor,
    Major,
}
```

Would then produce this:

```rust
impl Default for Variant {
    fn default() -> Variant {
        Variant::Undefined
    }
}
```

### PR DESCRIPTION
**Pull request by bnjjj**
close #6860

### CODE REVIEW

File: crates/assists/src/tests/generated.rs [hunk 1]
```diff
@@ -364,6 +364,33 @@ fn foo<T: Copy + Clone>() { }
...
+        r#####"
+enum Version {
+ Undefined,
+ Minor,
+ Major,
+}
+
+impl Default for Version {
+    fn default() -> Self {
+        Version::Minor
```
Comment (lnicola) [inline crates/assists/src/tests/generated.rs:387]:
```suggestion
        Self::Minor
```

File: crates/assists/src/handlers/generate_default_from_enum_variant.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    #[test]
+    fn test_generate_default_from_variant() {
+        check_assist(
+            generate_default_from_enum_variant,
+            r#"enum Variant {
+                Undefined,
+                Minor<|>,
+                Major,
+            }"#,
```
Comment (matklad) [inline crates/assists/src/handlers/generate_default_from_enum_variant.rs:112]: Let's remote indentation from test data:

https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#minimal-tests

Comment (bnjjj) [reply crates/assists/src/handlers/generate_default_from_enum_variant.rs]: updated

Comment (matklad) [reply crates/assists/src/handlers/generate_default_from_enum_variant.rs]: note that the example in the sylte guide also has a `\n` after `r#"` ;-)

Comment (bnjjj) [reply crates/assists/src/handlers/generate_default_from_enum_variant.rs]: oups sorry

**Comment by matklad** [comment]
bors r+

Looks great now!

**Pull request merged**